tests/workflows/test_from_csv_to_parquet.py

```diff
@@ -0,0 +1,140 @@
+
+
+@pytest.fixture
+def from_csv_to_parquet_client(
+    from_csv_to_parquet_cluster,
+    cluster_kwargs,
+    upload_cluster_dump,
+    benchmark_all,
+):
+    n_workers = cluster_kwargs["from_csv_to_parquet_cluster"]["n_workers"]
+    with Client(from_csv_to_parquet_cluster) as client:
+        from_csv_to_parquet_cluster.scale(n_workers)
+        client.wait_for_workers(n_workers)
+        client.restart()
+        with upload_cluster_dump(client), benchmark_all(client):
+            yield client
+
+
+COLUMNSV1 = OrderedDict([
+    ("GlobalEventID", "Int64"),
+    ("Day", "Int64"),
+    ("MonthYear", "Int64"),
+    ("Year", "Int64"),
+    ("FractionDate", "Float64"),
+    ("Actor1Code", "string[pyarrow]"),
+    ("Actor1Name", "string[pyarrow]"),
+    ("Actor1CountryCode", "string[pyarrow]"),
+    ("Actor1KnownGroupCode", "string[pyarrow]"),
+    ("Actor1EthnicCode", "string[pyarrow]"),
+    ("Actor1Religion1Code", "string[pyarrow]"),
+    ("Actor1Religion2Code", "string[pyarrow]"),
+    ("Actor1Type1Code", "string[pyarrow]"),
+    ("Actor1Type2Code", "string[pyarrow]"),
+    ("Actor1Type3Code", "string[pyarrow]"),
+    ("Actor2Code", "string[pyarrow]"),
+    ("Actor2Name", "string[pyarrow]"),
+    ("Actor2CountryCode", "string[pyarrow]"),
+    ("Actor2KnownGroupCode", "string[pyarrow]"),
+    ("Actor2EthnicCode", "string[pyarrow]"),
+    ("Actor2Religion1Code", "string[pyarrow]"),
+    ("Actor2Religion2Code", "string[pyarrow]"),
+    ("Actor2Type1Code", "string[pyarrow]"),
+    ("Actor2Type2Code", "string[pyarrow]"),
+    ("Actor2Type3Code", "string[pyarrow]"),
+    ("IsRootEvent", "Int64"),
+    ("EventCode", "string[pyarrow]"),
+    ("EventBaseCode", "string[pyarrow]"),
+    ("EventRootCode", "string[pyarrow]"),
+    ("QuadClass", "Int64"),
+    ("GoldsteinScale", "Float64"),
+    ("NumMentions", "Int64"),
+    ("NumSources", "Int64"),
+    ("NumArticles", "Int64"),
+    ("AvgTone", "Float64"),
+    ("Actor1Geo_Type", "Int64"),
+    ("Actor1Geo_Fullname", "string[pyarrow]"),
+    ("Actor1Geo_CountryCode", "string[pyarrow]"),
+    ("Actor1Geo_ADM1Code", "string[pyarrow]"),
+    ("Actor1Geo_Lat", "Float64"),
+    ("Actor1Geo_Long", "Float64"),
+    ("Actor1Geo_FeatureID", "string[pyarrow]"),
+    ("Actor2Geo_Type", "Int64"),
+    ("Actor2Geo_Fullname", "string[pyarrow]"),
+    ("Actor2Geo_CountryCode", "string[pyarrow]"),
+    ("Actor2Geo_ADM1Code", "string[pyarrow]"),
+    ("Actor2Geo_Lat", "Float64"),
+    ("Actor2Geo_Long", "Float64"),
+    ("Actor2Geo_FeatureID", "string[pyarrow]"),
+    ("ActionGeo_Type", "Int64"),
+    ("ActionGeo_Fullname", "string[pyarrow]"),
+    ("ActionGeo_CountryCode", "string[pyarrow]"),
+    ("ActionGeo_ADM1Code", "string[pyarrow]"),
+    ("ActionGeo_Lat", "Float64"),
+    ("ActionGeo_Long", "Float64"),
+    ("ActionGeo_FeatureID", "string[pyarrow]"),
+    ("DATEADDED", "Int64"),
+    ("SOURCEURL", "string[pyarrow]"),
+])
+
+
+def test_from_csv_to_parquet(from_csv_to_parquet_client, s3_factory, s3_url):
```

Comment:
this test appears to be a bit convoluted.

The ordinary read_csv should just be one of

* `dd.read_csv("s3://gdelt-open-data/events/")`
* `dd.read_csv(files)`

why is this back and forth conversion and the `download` function required?

Reply:
I'm mostly raising this in case it is related to the problem you were mentioning.